src/basic/process-util.c

```diff
@@ -1186,10 +1187,14 @@ pid_t getpid_cached(void) {
 
                 new_pid = raw_getpid();
 
-                if (__register_atfork(NULL, NULL, reset_cached_pid, __dso_handle) != 0) {
-                        /* OOM? Let's try again later */
-                        cached_pid = CACHED_PID_UNSET;
-                        return new_pid;
+                if (!installed) {
+                        if (__register_atfork(NULL, NULL, reset_cached_pid, __dso_handle) != 0) {
```

Comment:
FTR: it returns either 0 or -ENOMEM, so this conditional is correct.

Reply:
given it's semi-documented state (glibc doesn't document it, and LSB only very superficially) I didn't want to hardcode errors, hence I tried to make the error check as generic as possible, which is why I didn't want to check the actual numbers

Reply:
(but i figure i could add this as a comment)